cmd/ipfs/add_migrations.go

```diff
@@ -31,6 +31,9 @@ func addMigrations(ctx context.Context, node *core.IpfsNode, fetcher migrations.
 
 	for _, fetcher := range fetchers {
 		switch f := fetcher.(type) {
+		case nil:
+			// https://github.com/ipfs/go-ipfs/issues/8780
+			continue
```

Comment:
How do we end up with one of these?

Reply:
We probably don't – opened this draft PR  to run CI and then look closer.

I now suspect we don't have nil here,  but a `RetryFetcher` which wraps `NewHttpFetcher`:
https://github.com/ipfs/go-ipfs/blob/a6d04184062d1883f090d9661a4d8170a726cfc4/repo/fsrepo/migrations/migrations.go#L165

iiuc `switch`  over `fetcher` type interprets  `RetryFetcher` as something other than `IpfsFetcher` and `HttpFetcher` and that is why default gets executed. 

Quick demo: https://go.dev/play/p/MEKo40x1z0I

Reply:
@aschmahmann if we added  `RetryFetcher` some time later,  that probably introduced the bug. 

Since it is only used for HTTP, we could fix it this way:

```diff
- case *migrations.HttpFetcher:
+ case *migrations.HttpFetcher, *migrations.RetryFetcher:
```


Thoughts?

Reply:
Can we add something to the Fetcher interface for this so we can remove this brittle type switch altogether?

Reply:
Probably, but at the same time we won't be adding any new Fetcher types any time soon.
I'm merging this one-liner fix for now – we will be revisiting fetchers after we have car  export on gateways anyway, we can refactor things then.